src/shogun/machine/FeatureDispatchCRTP.h

```diff
@@ -0,0 +1,120 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Shubham Shukla
+ */
+
+#ifndef _FEATUREDISPATCHCRTP_H__
+#define _FEATUREDISPATCHCRTP_H__
+
+#include <shogun/lib/config.h>
+
+#include <shogun/features/DenseFeatures.h>
+#include <shogun/lib/common.h>
+#include <shogun/machine/LinearMachine.h>
+
+namespace shogun
+{
+	class CFeatures;
+	class CLabels;
+#define IGNORE_IN_CLASSLIST
+
+	template <class P, class T>
+	IGNORE_IN_CLASSLIST class CDenseFeaturesDispatch : public T
```

Comment:
the class name should match the filename please.
Or is your plan to put all the dispatcher classes into one file?

Reply:
yes that is the plan